### ISSUE TITLE
CartesianAxis example

### ISSUE THREAD

**Issue by sandbox4013** [recharts/recharts#2476]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
I want to draw custom CartesianAxis
Eg: 3 horizontal axis - first and last solid, middlle - dashed
CartesianGrid allows to style all axes

I can't handle how to use CartesianAxis - seems ticks array missing, but I'm unsure how to proper fill it

Could you create some example at official page?

### What does the proposed API look like?
Custom axes in examples
Don't use CartesianGrid(if it works that way, of course)

**Comment by DanTheOrange** [comment]
This issue is quite old, is there any work planned for this?

I've only found one proper reference of this component being used in this repo. [Here](https://github.com/recharts/recharts/blob/6c692d3919b75c348b9d4a034aca011fb40f0202/demo/component/CartesianAxis.tsx), but it isn't particularly useful. To me it looks like an internally used component only, but it is [clearly in the docs](https://recharts.org/en-US/api/CartesianAxis).

Any help is appreciated. 
Thanks

**Comment by ckifer** [comment]
Agreed with @DanTheOrange,

My understanding is that `CartesianAxis ` is only used internally in the repo. We'll try to understand and clean up the docs on this.

**Comment by sandbox4013** [comment]
> Agreed with @DanTheOrange,
> 
> My understanding is that `CartesianAxis ` is only used internally in the repo. We'll try to understand and clean up the docs on this.

Anyway, it’s good to have such a feature

**Comment by nikolasrieble** [comment]
Good point. Should we also not have stories in storybook for internal only components? 
It might still make sense to enable testing and development, but certainly we should mark them as such.

**Comment by ckifer** [comment]
should totally avoid adding internal only anything to the docs in the future. 

For this one, imo the given example is a good one - it looks like `ticks` is available for use.

[![Edit simple-cartesian-axis(forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/simple-cartesian-axis-forked-qxyhg4?fontsize=14&hidenavigation=1&theme=dark)

![image](https://github.com/recharts/recharts/assets/25180830/e04c652c-d89b-4c2f-8f49-55d5925eb033)
